core/src/main/java/io/opencensus/stats/StatsComponent.java

```diff
@@ -13,21 +13,58 @@
 
 package io.opencensus.stats;
 
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
 /**
  * Class that holds the implementations for {@link ViewManager}, {@link StatsRecorder}, and {@link
  * StatsContextFactory}.
  *
  * <p>All objects returned by methods on {@code StatsComponent} are cacheable.
  */
-// TODO(sebright): Add a no-op StatsComponent.
 public abstract class StatsComponent {
+  private static final StatsComponent NOOP_STATS_COMPONENT = new NoopStatsComponent();
+
   /** Returns the default {@link ViewManager}. */
+  @Nullable
   public abstract ViewManager getViewManager();
 
   /** Returns the default {@link StatsRecorder}. */
   public abstract StatsRecorder getStatsRecorder();
 
   /** Returns the default {@link StatsContextFactory}. */
   // TODO(sebright): Remove this method once StatsContext is replaced by TagContext.
+  @Nullable
   abstract StatsContextFactory getStatsContextFactory();
+
+  /**
+   * Returns a {@code StatsComponent} that has a no-op implementation for the {@link StatsRecorder}.
+   *
+   * @return a {@code StatsComponent} that has a no-op implementation for the {@code StatsRecorder}.
+   */
+  static StatsComponent getNoopStatsComponent() {
+    return NOOP_STATS_COMPONENT;
+  }
+
+  @Immutable
+  private static final class NoopStatsComponent extends StatsComponent {
+
+    @Override
+    @Nullable
+    public ViewManager getViewManager() {
+      // TODO(sebright): Decide whether to also provide a no-op implementation for the ViewManager.
+      return null;
+    }
+
+    @Override
+    public StatsRecorder getStatsRecorder() {
+      return StatsRecorder.getNoopStatsRecorder();
+    }
+
+    @Override
+    @Nullable
+    StatsContextFactory getStatsContextFactory() {
+      return null;
```

Comment:
getStatsContextFactory() is going to be removed when we replace StatsContext with TagContext, so I would rather leave it null until then.  We will need something similar in the `tags` package after #465 is merged, though.